lib/Sapi.php

```diff
@@ -113,6 +113,12 @@ public static function sendResponse(ResponseInterface $response): void
                     if ($copied <= 0) {
                         break;
                     }
+                    // Abort on client disconnect.
+                    // With ignore_user_abort(true), the script is not aborted on client disconnect.
+                    // To avoid reading the entire stream and dismissing the data afterward, check between the chunks if the client is still there.
+                    if (1 === ignore_user_abort() && 1 === connection_aborted()) {
```

Comment:
https://www.php.net/manual/en/function.ignore-user-abort.php
This PHP function does return an `int` - looking at its name, I would have assumed it returned a `bool` !
So this code does look OK.

Reply:
I had the same experience ;)

Reply:
also `connection_aborted` impurity will be covered by https://github.com/phpstan/phpstan-src/pull/2555